documentation/projects/proposals/trust_and_safety/content_report_moderation/20231208-implementation_plan_django_admin_moderator_access.md

```diff
@@ -0,0 +1,587 @@
+to view all the metadata for a work, we might as well take advantage of the
+existing admin view for media items that already has some of that built into it.
+I'll cover more details about how we'll accomplish this in the step-by-step.
+
+##### Actions
+
+To understand the rationale in this section, keep in mind that there are
+currently three report types (sensitive, DMCA, and "other"), and that users can
+select any of these. There is little guidance beyond the report type labels
+themselves to indicate to users which one to select. This means that reports of
+different types get mixed up, and especially the "other" type ends up with
+reports that would have been better suited as sensitive or DMCA reports. This
+means there is no coherent way to handle "all pending reports". A previous
+version of this implementation plan attempted to do that, but it's far too
+complex and has so many edge cases, that it's _easier_ to handle a set number of
+reports at a time, selected by the moderator.
+
+In the future, we could try to include links that open tables to show all
+moderation decisions for works by the same creator, etc, in case there is useful
+information, particularly for identifying potential bulk moderation actions.
+These kinds of additional views are out of scope for this current project,
+however.
+
+Each pending report should have a checkbox allowing it to be selected for
+actioning. If there is only a single pending report, it should be automatically
+selected when the form loads.
+
+The expanded view should have the following context-sensitive actions:
+
+- "Mark sensitive: mark the work as sensitive based on selected reports"
+  - Only available if a work is not already marked sensitive
+  - `confirmed_sensitive` action
+- "Deindex sensitive: entirely exclude the work from Openverse due to
+  sensitivity based on selected reports"
+  - There are no real conditions for this because deindexing can happen for
+    works that are or are not marked sensitive
+  - `deindexed_sensitive` action
+- "Deindex copyright: entirely exclude the work from Openverse due to copyright
+  based on selected reports"
+  - `deindexed_copyright`
+- "Reject reports: take no action and mark all selected reports as reviewed"
+  - `rejected_reports` action
+- "Mark selected reports duplicates: take no action and mark all selected
+  reports as duplicates"
+  - `duplicate_reports` action
+
+Each of these actions should create a single `ModerationDecision` with
+explanation, action, moderator ID, and so forth populated. The admin view should
+update the selected reports to set the `decision_id` to point to the new
+moderation decision. All actions rely on the selected set of reports and should
+not modify any other reports.
+
+Remove all logic from the report models `save` methods that bulk actions
+reports. That approach will no longer be valid and we will operate on specific
+reports individually following the `ModerationDecision`.
```

Comment:
Removing that logic from the `AbstractMediaReport` model, would that also imply removing its `status` column? It won't make sense with the current option values since the model will represent exclusively a user input, not the decision, and from what I understand if that a report will now have only two possible states: `pending` or `no_pending`.

To determine if a model if pending from decision can be a simply check of if the new `decision_id` column is null or not. Am I right this is what is expected here?

Reply:
That's true, we could get rid of the status column altogether and treat `decision_id`'s presence to indicate the reviewed state. That would simplify the data model a bit and even simplify https://github.com/WordPress/openverse/issues/1965 because we wouldn't have to migrate those fields anymore.

Reply:
Totally! If reports don't have another state, I can't see a more straightforward approach. I think we should go with it.

Reply:
Sounds good. The question is then how we back-fill Moderation Decision objects. We can backfill them, for each one:

- Create a moderation decision with moderation_id set to the `opener` django admin user
- Set the appropriate action on the decision based on the status of the report object
- Add an explanation on the decision indicating it is a backfill record

There are currently a small number of these, so it would be a fast backfill to complete in a simple management command.

Reply:
+1 to all this from me as well :) Especially since the backfill should be simple